### PR TITLE
Add cilium bpf nodeid list to bugtool and print nodeid in hex in ipcache dump

### PR DESCRIPTION
**Pull request by brb**
To help to detect when IPcache is out of sync with locally stored Node IDs.

### CODE REVIEW

**Review by gandro — APPROVED** [review]

**Review by pchaigno — APPROVED** [review]

**Comment by brb** [comment]
Sorry, pushed to the same PR other change (dump the nodeid in hex in the IPcache output)

**Review by aspsk — APPROVED** [review]

File: pkg/maps/ipcache/ipcache.go [hunk 1]
```diff
@@ -143,8 +144,8 @@ type RemoteEndpointInfo struct {
 }
 
 func (v *RemoteEndpointInfo) String() string {
-	return fmt.Sprintf("identity=%d encryptkey=%d tunnelendpoint=%s nodeid=%d",
-		v.SecurityIdentity, v.Key, v.TunnelEndpoint, v.NodeID)
+	return fmt.Sprintf("identity=%d encryptkey=%d tunnelendpoint=%s nodeid=0x%s",
+		v.SecurityIdentity, v.Key, v.TunnelEndpoint, strconv.FormatUint(uint64(v.NodeID), 16))
```
Comment (pchaigno) [inline pkg/maps/ipcache/ipcache.go:134]: Do we also want to update https://github.com/cilium/cilium/blob/c2c5ed3334ae8150fd06080c2a937ec9f3041964/cilium/cmd/bpf_nodeid_list.go#L84 and https://github.com/cilium/cilium/blob/c2c5ed3334ae8150fd06080c2a937ec9f3041964/cilium/cmd/node_id_list.go#L54?

**Review by asauber — APPROVED** [review]
Looks harmless enough. Leading zeroes would still be truncated, which seems fine.

File: pkg/maps/ipcache/ipcache.go (hunk 1 above)
Comment (brb) [reply pkg/maps/ipcache/ipcache.go:134]: Yep, good suggestion.

**Review by pchaigno — APPROVED** [review]

**Comment by pchaigno** [comment]
This isn't covered by end-to-end CI tests. It's reviewed and all other tests passed so marking ready to merge.

**Pull request merged**